crates/iceberg/src/spec/view_version.rs

```diff
@@ -0,0 +1,347 @@
+#[serde(from = "ViewVersionV1", into = "ViewVersionV1")]
+#[builder(field_defaults(setter(prefix = "with_")))]
+/// A view versions represents the definition of a view at a specific point in time.
+pub struct ViewVersion {
+    /// A unique long ID
+    version_id: i64,
+    /// ID of the schema for the view version
+    schema_id: SchemaId,
+    /// Timestamp when the version was created (ms from epoch)
+    timestamp_ms: i64,
+    /// A string to string map of summary metadata about the version
+    summary: HashMap<String, String>,
+    /// A list of representations for the view definition.
+    representations: ViewRepresentations,
+    /// Catalog name to use when a reference in the SELECT does not contain a catalog
+    #[builder(default = None)]
+    default_catalog: Option<String>,
+    /// Namespace to use when a reference in the SELECT is a single identifier
+    default_namespace: NamespaceIdent,
+}
+
+impl ViewVersion {
+    /// Get the version id of this view version.
+    #[inline]
+    pub fn version_id(&self) -> i64 {
+        self.version_id
+    }
+
+    /// Get the schema id of this view version.
+    #[inline]
+    pub fn schema_id(&self) -> SchemaId {
+        self.schema_id
+    }
+
+    /// Get the timestamp of when the view version was created
+    #[inline]
+    pub fn timestamp(&self) -> MappedLocalTime<DateTime<Utc>> {
+        Utc.timestamp_millis_opt(self.timestamp_ms)
+    }
+
+    /// Get the timestamp of when the view version was created in milliseconds since epoch
+    #[inline]
+    pub fn timestamp_ms(&self) -> i64 {
+        self.timestamp_ms
+    }
+
+    /// Get summary of the view version
+    #[inline]
+    pub fn summary(&self) -> &HashMap<String, String> {
+        &self.summary
+    }
+
+    /// Get this views representations
+    #[inline]
+    pub fn representations(&self) -> &ViewRepresentations {
+        &self.representations
+    }
+
+    /// Get the default catalog for this view version
+    #[inline]
+    pub fn default_catalog(&self) -> Option<&String> {
+        self.default_catalog.as_ref()
+    }
+
+    /// Get the default namespace to use when a reference in the SELECT is a single identifier
+    #[inline]
+    pub fn default_namespace(&self) -> &NamespaceIdent {
+        &self.default_namespace
+    }
+
+    /// Get the schema of this snapshot.
+    pub fn schema(&self, view_metadata: &ViewMetadata) -> Result<SchemaRef> {
+        let r = view_metadata
+            .schema_by_id(self.schema_id())
+            .ok_or_else(|| {
+                Error::new(
+                    ErrorKind::DataInvalid,
+                    format!("Schema with id {} not found", self.schema_id()),
+                )
+            })
+            .cloned();
+        r
+    }
+
+    pub(crate) fn log(&self) -> ViewVersionLog {
+        ViewVersionLog::new(self.version_id, self.timestamp_ms)
+    }
+}
+
+/// A list of view representations.
+pub type ViewRepresentations = Vec<ViewRepresentation>;
```

Comment:
is it normal in rust to create a separate type for a list of X?

Reply:
Normally not. This is looking forward to the ViewMetadataBuilder where we want to implement methods on those.
The motivation is that multiple `ViewRepresentations` are more than just a Vec<ViewRepresentation> - there are many rules in constructing it (i.e. no duplicate dialects). We shouldn't just expose the Vec interface externally because then users could just add an item to the list without going through any of our validation functions (which are not there yet, will come with the ViewMetadataBuilder).

Its a very valuable comment though, because I am still exposing the interface currently. I am putting it in a struct with a private field.

Reply:
Fix: https://github.com/c-thiel/iceberg-rust/commit/2b427e511dbeb3dfa6292af6983ee8d0b8bc48ad